app/models/history.rb

```diff
@@ -463,30 +466,42 @@ def self.clear_follow_up_flag(history)
   end
 
   private_class_method def self.compose_message(history, field)
-    creator = history[:household_status] == :patient ? 'User' : 'System'
+    creator = history[:initiator_id].nil? ? 'System' : 'User'
     verb = field[:new_value].blank? ? 'cleared' : 'changed'
     from_text = field[:old_value].blank? ? 'blank' : "\"#{field[:old_value]}\""
     to_text = field[:new_value].blank? ? 'blank' : "\"#{field[:new_value]}\""
 
     comment = "#{creator} #{verb} #{field[:name]} from #{from_text} to #{to_text}"
-    comment += compose_explanation(history, field)
+    comment += compose_explanation(history)
     comment += history[:note] unless history[:note].blank?
     comment += '.'
     comment += " Reason: #{history[:reason]}" unless history[:reason].blank?
     comment
   end
 
-  private_class_method def self.compose_explanation(history, field)
-    if history[:household_status] == :patient && history[:propagation] == :group
-      " and chose to update this #{field[:type]} for all household members"
-    elsif history[:household_status] == :patient && history[:propagation] == :group_cm
-      " and chose to update this #{field[:type]} for household members under continuous exposure"
-    elsif history[:household_status] != :patient && history[:propagation] == :group
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for all household members"
-    elsif history[:household_status] != :patient && history[:propagation] == :group_cm
-      " because User updated #{field[:name]} for another member in this monitoree's household and chose to update this
-        #{field[:type].nil? ? 'field' : field[:type]} for household members under continuous exposure"
+  private_class_method def self.compose_explanation(history)
+    if history[:initiator_id] == history[:patient].id && history[:propagation] == :group
+      ' and applied that change to all household members'
+    elsif history[:initiator_id] == history[:patient].id && history[:propagation] == :group_cm
+      ' and applied that change to household members under continuous exposure'
```

Comment:
How do we get this history message? I can't quite figure out how to make this happen, and can't see where we use `group_cm` outside of this function.

Reply:
You're right, it's not currently possible: I included it as an option because it was there in the original function. My guess is that when this code was originally written there was an interest in having that specificity in Continuous Exposure cases but it was left as a todo. I'm fine with removing it, but I'll want to first check if there's a backlog item about this somewhere to make sure I'm not making a team member's job harder at some point in the future.

Reply:
Yeah I've tried to figure out in some older versions if this was used, but I'm not seeing it, but then again maybe I'm missing something. @holmesie, do you know why the `:group_cm` option is there and where (if) it gets used?

Reply:
Per some slack conversation, it sounds like we are agreed this isn't used, and it can go.